mempool/v1/mempool.go

```diff
@@ -713,34 +708,23 @@ func (txmp *TxMempool) recheckTransactions() {
 
 	// Issue CheckTx calls for each remaining transaction, and when all the
 	// rechecks are complete signal watchers that transactions may be available.
-	go func() {
-		g, start := taskgroup.New(nil).Limit(2 * runtime.NumCPU())
```

Comment:
do you think it would be possible to avoid this issue without blocking by simply not using a taskgroup?

Reply:
To be watertight it should be blocking over the period that state is updating else as this is still a go routine,  transactions may be submitted before updating to the latest nonce with the checktxstate

Reply:
isn't the mempool locked while the goroutine is still running?

Reply:
no it's not

Reply:
It could happen in practice that some of these CheckTx transactions happen while the mempool is locked but it's not guaranteed